File: Python/services/text_recognizers_client.py
from recognizers_text import Culture, ModelResult
from recognizers_choice import ChoiceRecognizer
from recognizers_date_time import DateTimeRecognizer 
from recognizers_number import NumberRecognizer
from recognizers_number_with_unit import NumberWithUnitRecognizer 
from recognizers_sequence import SequenceRecognizer 

class text_recognizers_client:
    def __init__(self, cultures, recognizers):
        self.models = []
        
        print(cultures)
        print(recognizers)

        for culture in cultures:
            if 'Choice' in recognizers:
                self.add_choice_models(culture)

            if 'DateTime' in recognizers:
                self.add_date_time_models(culture)

            if 'Number' in recognizers:
                self.add_number_recognizers_models(culture)
            
            if 'NumberWithUnit' in recognizers:
                self.add_number_with_unit_recognizers_models(culture)

            if 'Sequence' in recognizers:
                self.add_sequence_models(culture)
        
        

    def add_choice_models(self, culture):
        choice_recognizer = ChoiceRecognizer(culture)
        self.models.append(choice_recognizer.get_boolean_model())

    def add_date_time_models(self, culture):
        date_time_recognizer = DateTimeRecognizer(culture)
        self.models.append(date_time_recognizer.get_datetime_model())

    def add_number_recognizers_models(self, culture):
        print(culture)
        number_recognizer = NumberRecognizer(culture)
        self.models.append(number_recognizer.get_number_model())
        self.models.append(number_recognizer.get_ordinal_model())
        self.models.append(number_recognizer.get_percentage_model())

    def add_number_with_unit_recognizers_models(self, culture):
        number_with_unit_recognizer = NumberWithUnitRecognizer(culture)
        self.models.append(number_with_unit_recognizer.get_age_model())
        self.models.append(number_with_unit_recognizer.get_currency_model())
        self.models.append(number_with_unit_recognizer.get_dimension_model())
        self.models.append(number_with_unit_recognizer.get_temperature_model())

    def add_sequence_models(self, culture):
        sequence_recognizer = SequenceRecognizer(culture)
        self.models.append(sequence_recognizer.get_phone_number_model())
        self.models.append(sequence_recognizer.get_ip_address_model())
        self.models.append(sequence_recognizer.get_mention_model())
        self.models.append(sequence_recognizer.get_hashtag_model())
        self.models.append(sequence_recognizer.get_url_model())
        self.models.append(sequence_recognizer.get_guid_model())
        self.models.append(sequence_recognizer.get_email_model())

    def run_test(self, input):
        for model in self.models:
            print(model.parse(input))

File: .Net/RecognizersTextPerformanceTest/Services/TextRecognizersClient.cs
using Common.enums;
using Microsoft.Recognizers.Text;
using Microsoft.Recognizers.Text.Choice;
using Microsoft.Recognizers.Text.DateTime;
using Microsoft.Recognizers.Text.Number;
using Microsoft.Recognizers.Text.NumberWithUnit;
using Microsoft.Recognizers.Text.Sequence;
using RecognizersTextPerformanceTest.Interfaces;
using System.Collections.Generic;

namespace RecognizersTextPerformanceTest.Services
{
    public class TextRecognizersClient : ITextRecognizerClient
    {
        private List<IModel> _models;
        public TextRecognizersClient(string culture, Recognizers recognizer)
        {
            _models = new List<IModel>();

            switch (recognizer)
            {
                case Recognizers.Choice:
                    AddChoiceModels(culture);
                    break;
                case Recognizers.DateTime:
                    AddDateTimeModels(culture);
                    break;
                case Recognizers.Number:
                    AddNumberModels(culture);
                    break;
                case Recognizers.NumberWithUnit:
                    AddNumberWithUnitModels(culture);
                    break;
                case Recognizers.Sequence:
                    AddSequenceModels(culture);
                    break;
            }

        }
        private void AddChoiceModels(string culture)
        {
            var choiceRecognizr = new ChoiceRecognizer(culture);
            _models.Add(choiceRecognizr.GetBooleanModel());

            choiceRecognizr = null;
        }

        private void AddDateTimeModels(string culture)
        {
            var dateTimeRecognizer = new DateTimeRecognizer(culture);
            _models.Add(dateTimeRecognizer.GetDateTimeModel());

            dateTimeRecognizer = null;
        }

        private void AddNumberModels(string culture)
        {
            var numberRecognizer = new NumberRecognizer(culture);
            _models.Add(numberRecognizer.GetNumberModel());
            _models.Add(numberRecognizer.GetNumberRangeModel());
            _models.Add(numberRecognizer.GetOrdinalModel());
            _models.Add(numberRecognizer.GetPercentageModel());

            numberRecognizer = null;
        }

        private void AddNumberWithUnitModels(string culture)
        {
            var numberWithUnitRecognizer = new NumberWithUnitRecognizer(culture);
            _models.Add(numberWithUnitRecognizer.GetAgeModel());
            _models.Add(numberWithUnitRecognizer.GetCurrencyModel());
            _models.Add(numberWithUnitRecognizer.GetDimensionModel());
            _models.Add(numberWithUnitRecognizer.GetTemperatureModel());

            numberWithUnitRecognizer = null;
        }

        private void AddSequenceModels(string culture)
        {
            var sequenceRecognizer = new SequenceRecognizer(culture);
            _models.Add(sequenceRecognizer.GetEmailModel());
            _models.Add(sequenceRecognizer.GetGUIDModel());
            _models.Add(sequenceRecognizer.GetHashtagModel());
            _models.Add(sequenceRecognizer.GetIpAddressModel());
            _models.Add(sequenceRecognizer.GetMentionModel());
            _models.Add(sequenceRecognizer.GetPhoneNumberModel());
            _models.Add(sequenceRecognizer.GetQuotedTextModel());

            sequenceRecognizer = null;
        }
        public void RunTest(string test)
        {
            foreach (var model in _models)
                model.Parse(test);
        }
    }
}
